src/encoded/static/components/genome_browser.js

```diff
@@ -43,543 +47,482 @@ const dummyFiles = [
-                desc: 'C. elegans reference genome build ce10/WS220',
-                tier_type: 'sequence',
-                provides_entrypoints: true,
-                pinned: true,
-            },
-            {
-                name: 'Genes',
-                desc: 'Gene structures from UCSC ce10/WS220',
-                bwgURI: 'https://s3-us-west-1.amazonaws.com/encoded-build/browser/ce10/ce10_UCSC_RefSeq_track.bb',
-                // stylesheet_uri: 'https://s3-us-west-1.amazonaws.com/encoded-build/browser/ce10/',
-                collapseSuperGroups: true,
-                trixURI: 'https://s3-us-west-1.amazonaws.com/encoded-build/browser/ce10/ce10_UCSC_RefSeq_track.ix',
-            },
-            {
-                name: 'Repeats',
-                desc: 'Repeat annotation from UCSC',
-                bwgURI: 'https://s3-us-west-1.amazonaws.com/encoded-build/browser/ce10/repeats_ce10.bb',
-                // stylesheet_uri: 'https://s3-us-west-1.amazonaws.com/encoded-build/browser/ce10/',
-                forceReduction: -1,
-            },
-        ];
+function mapGenome(inputAssembly) {
+    let genome = inputAssembly;
+    if (inputAssembly === 'hg19') {
+        genome = 'GRCh37';
+    } else if (inputAssembly === 'mm9') {
+        genome = 'GRCm37';
+    } else if (inputAssembly === 'mm10') {
+        genome = 'GRCm38';
     }
-    return browserCfg;
+    return genome;
 }
 
-
 class GenomeBrowser extends React.Component {
     constructor() {
         super();
 
-        // Bind this to non-React methods.
-        this.locationChange = this.locationChange.bind(this);
-        this.makeTrackLabel = this.makeTrackLabel.bind(this);
+        this.state = {
+            width: 592,
+            trackList: [],
+            visualizer: null,
+            showAutoSuggest: true,
+            searchTerm: '',
+            genome: '',
+            contig: 'chr1',
+            x0: 0,
+            x1: 59e6,
+            pinnedFiles: [],
+        };
+        this.setBrowserDefaults = this.setBrowserDefaults.bind(this);
+        this.filesToTracks = this.filesToTracks.bind(this);
+        this.drawTracks = this.drawTracks.bind(this);
+        this.drawTracksResized = this.drawTracksResized.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleAutocompleteClick = this.handleAutocompleteClick.bind(this);
+        this.handleOnFocus = this.handleOnFocus.bind(this);
     }
 
     componentDidMount() {
-        const { assembly, region, limitFiles } = this.props;
-        // console.log('DidMount ASSEMBLY: %s', assembly);
-
-        // Probably not worth a define in globals.js for visualizable types and statuses.
+        const genome = mapGenome(this.props.assembly);
+        this.setState({ genome });
+        const genomePromise = new Promise((resolve) => {
+            this.setBrowserDefaults(genome);
+            resolve('success!');
+        });
```

Comment:
I coded it this way because `setBrowserDefaults` sets the "pinnedFiles" (the genome and gene files that are displayed at the top of the genome browser), and I needed to make sure these files are in state before I try to use them in the next couple steps where I convert the files to tracks and display the tracks on the browser. It's very complicated the way everything is coded because there are a lot of things stored in state that need to be updated before the next thing that relies on that update can be updated, but I couldn't think of a simpler solution.

Reply:
Looking at this more, I think you are right and `resolve('success')` should be in `setBrowserDefaults`.